flytekit/core/python_third_party_task.py

```diff
@@ -0,0 +1,212 @@
+        return self._task_resolver
+
+    @property
+    def task_template(self) -> Optional[_task_model.TaskTemplate]:
+        """
+        Override the base class implementation to serialize on first call.
+        """
+        return self._task_template or self.serialize_to_model(settings=PythonThirdPartyContainerTask.SERIALIZE_SETTINGS)
+
+    @property
+    def container_image(self) -> str:
+        return self._container_image
+
+    def get_command(self, settings: SerializationSettings) -> List[str]:
+        container_args = [
+            "pyflyte-execute",
+            "--inputs",
+            "{{.input}}",
+            "--output-prefix",
+            "{{.outputPrefix}}",
+            "--raw-output-data-prefix",
+            "{{.rawOutputDataPrefix}}",
+            "--resolver",
+            self.task_resolver.location,
+            "--",
+            *self.task_resolver.loader_args(settings, self),
+        ]
+
+        return container_args
+
+    def get_container(self, settings: SerializationSettings) -> _task_model.Container:
+        env = {**settings.env, **self.environment} if self.environment else settings.env
+        return _get_container_definition(
+            image=self.container_image,
+            command=[],
+            args=self.get_command(settings=settings),
+            data_loading_config=None,
+            environment=env,
+            storage_request=self.resources.requests.storage,
+            cpu_request=self.resources.requests.cpu,
+            gpu_request=self.resources.requests.gpu,
+            memory_request=self.resources.requests.mem,
+            storage_limit=self.resources.limits.storage,
+            cpu_limit=self.resources.limits.cpu,
+            gpu_limit=self.resources.limits.gpu,
+            memory_limit=self.resources.limits.mem,
+        )
+
+    def serialize_to_model(self, settings: SerializationSettings) -> _task_model.TaskTemplate:
+        # This doesn't get called from translator unfortunately. Will need to move the translator to use the model
+        # objects directly first.
+        # Note: This doesn't settle the issue of duplicate registrations. We'll need to figure that out somehow.
+        # TODO: After new control plane classes are in, promote the template to a FlyteTask, so that authors of
+        #  third-party tasks have a familiar thing to work with.
+        obj = _task_model.TaskTemplate(
+            identifier_models.Identifier(
+                identifier_models.ResourceType.TASK, settings.project, settings.domain, self.name, settings.version
+            ),
+            self.task_type,
+            self.metadata.to_taskmetadata_model(),
+            self.interface,
+            self.get_custom(settings),
+            container=self.get_container(settings),
+            config=self.get_config(settings),
+        )
+        self._task_template = obj
+        return obj
+
+
+class TaskTemplateResolver(TrackedInstance, TaskResolverMixin):
```

Comment:
docs?

Reply:
why this rant
```This class mostly follows the TaskResolverMixin pattern but kinda doesn't as well. The key difference is that
    a task resolver was supposed to
    * Be able to restore the same task when ``load_task`` is called as the object that ``loader_args`` was called on.
      That is, even though at run time it's in a container on a cluster and is obviously a different Python process,
      the Python object in memory should look the same.
    * Offer a one-to-one mapping between the list of strings returned by the ``loader_args`` function, an the task,
      at least within the container.```

Do we have this contract explicitly spelled out somewhere - if not this seems like a not so useful comment

Reply:
what about - special resolver that resolves the ThirdPartyTask at runtime, using only the tasktemplate. Thus for any task, derived from x, this resolver is used and such tasks should be fully rehydrated from the task template alone

Reply:
i didn't think it was a rant.  i'll add this comment and change the wording a bit.